server/lib/src/plugins/ecdhkeygen.rs

```diff
@@ -0,0 +1,142 @@
+        _me: &crate::event::ModifyEvent,
+    ) -> Result<(), kanidm_proto::v1::OperationError> {
+        admin_error!("plugin {} has an unimplemented pre_modify!", Self::id());
+        Err(kanidm_proto::v1::OperationError::InvalidState)
+    }
+
+    fn pre_batch_modify(
+        _qs: &mut crate::server::QueryServerWriteTransaction,
+        _pre_cand: &[std::sync::Arc<crate::prelude::EntrySealedCommitted>],
+        _cand: &mut Vec<crate::prelude::EntryInvalidCommitted>,
+        _me: &crate::server::batch_modify::BatchModifyEvent,
+    ) -> Result<(), kanidm_proto::v1::OperationError> {
+        admin_error!(
+            "plugin {} has an unimplemented pre_batch_modify!",
+            Self::id()
+        );
+        Err(kanidm_proto::v1::OperationError::InvalidState)
+    }
+
+    fn pre_repl_refresh(
+        _qs: &mut crate::server::QueryServerWriteTransaction,
+        _cand: &[crate::prelude::EntryRefreshNew],
+    ) -> Result<(), kanidm_proto::v1::OperationError> {
+        admin_error!(
+            "plugin {} has an unimplemented pre_repl_refresh!",
+            Self::id()
+        );
+        Err(kanidm_proto::v1::OperationError::InvalidState)
+    }
+
+    fn post_repl_refresh(
+        _qs: &mut crate::server::QueryServerWriteTransaction,
+        _cand: &[crate::prelude::EntrySealedCommitted],
+    ) -> Result<(), kanidm_proto::v1::OperationError> {
+        admin_error!(
+            "plugin {} has an unimplemented post_repl_refresh!",
+            Self::id()
+        );
+        Err(kanidm_proto::v1::OperationError::InvalidState)
+    }
+
+    fn pre_repl_incremental(
+        _qs: &mut crate::server::QueryServerWriteTransaction,
+        _cand: &mut [(
+            crate::prelude::EntryIncrementalCommitted,
+            std::sync::Arc<crate::prelude::EntrySealedCommitted>,
+        )],
+    ) -> Result<(), kanidm_proto::v1::OperationError> {
+        admin_error!(
+            "plugin {} has an unimplemented pre_repl_incremental!",
+            Self::id()
+        );
+        // debug_assert!(false);
+        // Err(OperationError::InvalidState)
+        Ok(())
+    }
+
+    fn post_repl_incremental(
+        _qs: &mut crate::server::QueryServerWriteTransaction,
+        _pre_cand: &[std::sync::Arc<crate::prelude::EntrySealedCommitted>],
+        _cand: &[crate::prelude::EntrySealedCommitted],
+    ) -> Result<(), kanidm_proto::v1::OperationError> {
+        admin_error!(
+            "plugin {} has an unimplemented post_repl_incremental!",
+            Self::id()
+        );
+        // debug_assert!(false);
+        // Err(OperationError::InvalidState)
+        Ok(())
+    }
+
+    // fn verify(
+    //     _qs: &mut crate::server::QueryServerReadTransaction,
+    // ) -> Vec<Result<(), kanidm_proto::v1::ConsistencyError>> {
+    //     admin_error!("plugin {} has an unimplemented verify!", Self::id());
+    //     vec![Err(ConsistencyError::Unknown)]
+    // }
+}
```

Comment:
Tests?

Reply:
yeah I wrote the plugin on the fly just to see how the main tests were doing, but I still need to thoroughly think about what the plugin is supposed to do